server/terraform/modules/myhealth/main.tf

```diff
@@ -257,3 +264,57 @@ resource "google_compute_backend_service" "backend" {
   health_checks = null
   depends_on    = [google_project.project]
 }
+
+
+# Google Cloud bucket for static content assets.
+resource "google_storage_bucket" "content" {
+  name          = local.content_bucket_id
+  location      = var.bucket_location
+  force_destroy = false ### ???? The documentation on this is atrocious. I believe if this is false it will fail on destruction if there is data in the bucket?
```

Comment:
Put it at the end of the resource but this acts to prevent destruction of the resource which is what I think you want. Many other resources can be deleted and recreated without as much of an issue. If someone removes the `lifecyle` section, e.g. by deleting the whole resource, then it won't protect it.... but at least it acts as a prompt for accidental changes or a note in the code that someone should be extra careful.
```suggestion
  lifecycle {
    prevent_destroy = true
  }
```

Reply:
will do both. i believe force_destroy = false means it will fail on destroy if the bucket is non-empty; force_destroy=true means it'll zap the contents. but the documentation has so many double negatives and clauses with uncertain self-references.

Reply:
note that the prevent_destroy will block if the bucket is renamed (like i am doing now)... probably a good thing, certainly can be worked around and remind the operator they're doing something very strange.